rules/validator.go

```diff
@@ -164,11 +169,17 @@ func (v *validator) validateFilters(f map[string]string) error {
 }
 
 func (v *validator) validateRollupTags(ruleName string, tags []string) error {
-	requiredTags := v.opts.RequiredRollupTags()
-	if len(requiredTags) == 0 {
-		return nil
+	// Validate that all tags have valid characters
+	invalidChars := v.opts.RollupTagInvalidChars()
+	for _, tag := range tags {
+		fmt.Println("in here")
+		if err := validateChars(tag, invalidChars); err != nil {
+			return fmt.Errorf("invalid rollup tag %s for rollup rule %s : %s", tag, ruleName, err.Error())
+		}
 	}
+
 	// Validating the list of rollup tags in the rule contain all required tags.
+	requiredTags := v.opts.RequiredRollupTags()
```

Comment:
Perhaps still shortcut it by checking `len(requiredTags) == 0` and returning early if so to avoid allocating the map below?

Reply:
Yeah, good point. I also enclosed the invalid chars logic in an if len(invalidChars) > 0 clause - might be a little ugly but avoids unnecessary iteration through the tags. Let me know if you there's a better way.